src/main/java/org/lanternpowered/server/game/registry/type/block/BlockRegistryModule.java

```diff
@@ -862,16 +866,87 @@ public void registerDefaults() {
                         .translation("tile.chest.name")
                         .build("minecraft", "chest"),
                 this::chestData);
+        ///////////////////////
+        ///   Diamond Ore   ///
+        ///////////////////////
+        register(56, simpleBuilder()
+                        .itemType()
+                        .properties(builder -> builder
+                                .add(hardness(3.0))
+                                .add(blastResistance(15.0)))
+                        .translation("tile.oreDiamond.name")
+                        .build("minecraft", "diamond_ore"));
+        /////////////////////////
+        ///   Diamond Block   ///
+        /////////////////////////
+        register(57, simpleBuilder()
+                        .itemType()
+                        .properties(builder -> builder
+                                .add(hardness(5.0))
+                                .add(blastResistance(30.0)))
+                        .translation("tile.blockDiamond.name")
+                        .build("minecraft", "diamond_block"));
+        ////////////////
+        ///   Ice   ///
+        ///////////////
+        register(79, simpleBuilder()
+                        .itemType()
+                        .properties(builder -> builder
+                                .add(hardness(0.5))
+                                .add(blastResistance(2.5)))
+                        .translation("tile.ice.name")
+                        .build("minecraft", "ice"));
+        /////////////////
+        ///   Snow   ///
+        ////////////////
+        register(80, simpleBuilder()
+                        .itemType()
+                        .properties(builder -> builder
+                                .add(hardness(0.2))
+                                .add(blastResistance(1.0)))
+                        .translation("tile.snow.name")
+                        .build("minecraft", "snow"));
+        /////////////////
+        ///   Clay   ///
+        ////////////////
+        register(82, simpleBuilder()
+                        .itemType()
+                        .properties(builder -> builder
+                                .add(hardness(0.6))
+                                .add(blastResistance(3.0)))
+                        .translation("tile.clay.name")
+                        .build("minecraft", "clay"));
         ////////////////////
         ///    Pumpkin   ///
         ////////////////////
         register(86, pumpkinBuilder()
                         .itemType(builder -> builder
-                                .properties(properties -> properties
+                            .properties(properties -> properties
```

Comment:
There is some weird tabbing within the changes/additions you made, I will just comment a few with `+4 spaces` or `-4 spaces` where needed.